sql/src/main/java/com/vmware/ddlog/TranslateCreateTableDialect.java

```diff
@@ -22,6 +24,15 @@ protected String visitCreateTable(final CreateTable node, final String sql) {
         for (final TableElement element : node.getElements()) {
             if (element instanceof ColumnDefinition) {
                 final ColumnDefinition cd = (ColumnDefinition) element;
+                String h2Type = cd.getType();
+
+                // We need to strip the array subtype for H2
+                final Pattern arrayType = Pattern.compile("ARRAY\\((.+)\\)");
+                Matcher m = arrayType.matcher(cd.getType());
```

Comment:
you already have h2Type, you should use it here.

Reply:
fixed